src/kbc_modules/cc_kbc/mod.rs

```diff
@@ -39,8 +39,30 @@ impl KbcInterface for Kbc {
         Err(anyhow!("Check API of this KBC is unimplemented."))
     }
 
-    async fn decrypt_payload(&mut self, _annotation: &str) -> Result<Vec<u8>> {
-        Err(anyhow!("Decrypt Payload API of this KBC is unimplemented."))
+    async fn decrypt_payload(&mut self, annotation: &str) -> Result<Vec<u8>> {
+        let annotation_packet: AnnotationPacket = serde_json::from_str(annotation)
+            .map_err(|e| anyhow!("Failed to parse annotation: {}", e))?;
+
+        let key_url = annotation_packet.key_url;
+        if !key_url.starts_with(&self.kbs_uri) {
+            bail!("Muti-KBS resource is not supported")
+        }
+
+        let response = self.request_kbs_resource(key_url).await?;
+        let key = self.decrypt_response_output(response)?;
+
+        let iv = base64::decode(annotation_packet.iv)
+            .map_err(|e| anyhow!("Failed to decode initialization vector: {}", e))?;
+        let wrapped_data = base64::decode(annotation_packet.wrapped_data)
+            .map_err(|e| anyhow!("Failed to decode wrapped key: {}", e))?;
+
+        match annotation_packet.wrap_type.as_str() {
+            crypto::AES_GCM_256_ALGORITHM => crypto::aes_gcm_256_decrypt(&wrapped_data, &key, &iv)
+                .map_err(|e| anyhow!("Failed to decrypt annotation: {}", e)),
```

Comment:
Here we use AES-GCM-256 (which I think is good) but in online_sev_kbc we use openssl's AES-256-CTR:

https://github.com/confidential-containers/attestation-agent/blob/main/src/kbc_modules/online_sev_kbc/mod.rs#L60-L69

@fitzthum : I understand that offline_fs_kbc is used on s390x and it will gain from OpenSSL's optimizations. Why do we need OpenSSL and AES-256-CTR in the other KBCs?

Reply:
There is something I'm working on that might be related. As openssl is good for performance on s390x, I made [two features of crypto impls](https://github.com/confidential-containers/attestation-agent/pull/117/files#diff-899d7ab1242a214d596c21f0dda8d4df6696ba8186d7d931f253359d0ef0b78aR18) (purely rust and openssl), which can be selected by user when compiles AA.

I also found that [openssl supports AES-GCM](https://github.com/confidential-containers/attestation-agent/pull/117/files#diff-58cd7a44a49dd41a612e3eb7c58f89c41ae041b6604105abc2b3481f89f13d4eR39), but a different api from AES-CTR. I wonder whether we can use AES-GCM in offline-kbc?

Reply:
I think `online_sev_kbc` just uses the same cryto stuff as the offline KBC out of convenience. There has  been some discussion of unifying all of this so that container images are truly interchangeable, which I think would be a good idea.